src/store/mod.rs

```diff
@@ -134,40 +130,45 @@ pub struct StoreConfig {
     /// optional, and unused internally.
     pub size: Option<usize>,
 
-    /// The number of merkle tree levels above the base to cache on disk.
-    pub levels: usize,
+    /// The number of merkle tree rows_to_discard then cache on disk.
+    pub rows_to_discard: usize,
 }
 
 impl StoreConfig {
-    pub fn new<T: Into<PathBuf>, S: Into<String>>(path: T, id: S, levels: usize) -> Self {
+    pub fn new<T: Into<PathBuf>, S: Into<String>>(path: T, id: S, rows_to_discard: usize) -> Self {
         StoreConfig {
             path: path.into(),
             id: id.into(),
             size: None,
-            levels,
+            rows_to_discard,
         }
     }
 
     // If the tree is large enough to use the default, use it.  If
     // it's too small to cache anything, don't cache anything.
-    // Otherwise, the tree is 'small' so a fixed value of 2 levels
-    // above the base should be sufficient.
-    pub fn default_cached_above_base_layer(leafs: usize, branches: usize) -> usize {
+    // Otherwise, the tree is 'small' so a fixed value of height - 2
+    // rows_to_discard should be sufficient.
+    pub fn default_rows_to_discard(leafs: usize, branches: usize) -> usize {
         let height = get_merkle_tree_height(leafs, branches);
         if height <= 2 {
+            // If a tree only has a root row and/or base, there is
+            // nothing to discard.
```

Comment:
If this comment is accurate, then you may not be using 'height' as I expect. It's unclear whether this is a naming problem or a code-correctness problem (matter of perspective). While we are rooting out these naming-derived off-by-one errors, we should sort this one out too.

A tree with just a base row and a root (example: a binary tree with two base elements), has height = 1.

So a tree with height = 2 has exactly one intermediate row between the base and the root. This could be discarded, right?

Reply:
The height terminology above and implementation differ, and height is used consistently as including the base in the impl (so a tree with just a base row and a root has height 2 -- a holdover from the original fork).

Reply:
After discussion, it's clear that the implementation only considers 'row_count', so I've replaced the incorrect usages with that (and added a comment about this).

Reply:
^ Thanks for the suggested naming by the way :+1: